ext/standard/uuid.c

```diff
@@ -0,0 +1,662 @@
+{
+	return throw_uuid_parse_exception(
+		input,
+		input_len,
+		position,
+		"Expected at least %u hexadecimal digits, but got %u",
+		UUID_HEX_LEN,
+		actual
+	);
+}
+
+static zend_always_inline zend_object *throw_uuid_parse_exception_invalid_char(const char *input, const size_t input_len, const size_t position)
+{
+	return throw_uuid_parse_exception(
+		input,
+		input_len,
+		position,
+		"Expected hexadecimal digit, but found '%c' (0x%02x)",
+		input[position],
+		input[position]
+	);
+}
+
+PHPAPI int php_uuid_parse(php_uuid *uuid, const char *input, const size_t input_len, const zend_bool throw)
+{
+	size_t position = 0;
+	size_t digit    = 0;
+	size_t limit    = input_len - 1;
+	uint8_t byte    = 0;
+
+	while (input[position] == '-' || input[position] == ' ' || input[position] == '\t' || input[position] == '{') {
+		++position;
+	}
+
+	if (memcmp(input + position, URN_PREFIX, URN_PREFIX_LEN) == 0) {
+		position += URN_PREFIX_LEN;
+	}
+
+	while (input[limit] == '-' || input[limit] == ' ' || input[limit] == '\t' || input[limit] == '}') {
+		--limit;
+	}
+
+	if ((limit - position + 1) < UUID_HEX_LEN) {
+		if (throw) {
+			throw_uuid_parse_exception_invalid_len(input, input_len, position, limit - position + 1);
+		}
+		return FAILURE;
+	}
+
+	for (; position <= limit; ++position) {
+		const char chr = input[position];
+
+		/* First digit of the byte. */
+		if (digit % 2 == 0) {
+			if ('0' <= chr && chr <= '9') {
+				byte = chr - '0';
+			}
+			else if ('a' <= chr && chr <= 'f') {
+				byte = chr - 'a' + 10;
+			}
+			else if ('A' <= chr && chr <= 'F') {
+				byte = chr - 'A' + 10;
+			}
+			else if (chr == '-') {
+				continue;
+			}
+			else {
+				if (throw) {
+					throw_uuid_parse_exception_invalid_char(input, input_len, position);
+				}
+				return FAILURE;
+			}
+		}
+		/* Second digit of the byte. */
+		else {
+			/* Shift upper half. */
+			byte *= 16;
+
+			if ('0' <= chr && chr <= '9') {
+				byte += chr - '0';
+			}
+			else if ('a' <= chr && chr <= 'f') {
+				byte += chr - 'a' + 10;
+			}
+			else if ('A' <= chr && chr <= 'F') {
+				byte += chr - 'A' + 10;
+			}
+			else if (chr == '-') {
+				continue;
+			}
+			else {
+				if (throw) {
+					throw_uuid_parse_exception_invalid_char(input, input_len, position);
+				}
+				return FAILURE;
+			}
+
+			if (digit > UUID_HEX_LEN) {
+				if (throw) {
+					throw_uuid_parse_exception(
+						input,
+						input_len,
+						position,
+						"Expected no more than %u hexadecimal digits",
+						UUID_HEX_LEN
+					);
+				}
+				return FAILURE;
+			}
+
+			uuid->bytes[digit / 2] = byte;
+		}
+
+		++digit;
+	}
+
+	if (digit < UUID_HEX_LEN) {
+		if (throw) {
+			throw_uuid_parse_exception_invalid_len(input, input_len, position, digit);
+		}
+		return FAILURE;
+	}
+
+	return SUCCESS;
+}
+
+static zend_always_inline php_uuid *get_uuid(/*const*/ zval *uuid_object)
+{
+	zval *bytes = zend_read_property(php_ce_UUID, uuid_object, UUID_BYTES_PROP, UUID_BYTES_PROP_LEN, 1, NULL);
```

Comment:
Any particular reason to make this silent?

Reply:
Even if it's not going to be necessary in this specific case, an rv zval should still be passed.